docs/src/main/asciidoc/security-oidc-auth0-tutorial.adoc

```diff
@@ -0,0 +1,587 @@
+[id="security-oidc-auth0-tutorial"]
+= Protect Quarkus web application by using Auth0 OpenID Connect provider
+include::_attributes.adoc[]
+:diataxis-type: tutorial
+:categories: security,web
+
+xref:security-architecture.adoc[Quarkus Security] provides comprehensive OpenId Connect (OIDC) and OAuth2 support with its `quarkus-oidc` extension, supporting both xref:security-oidc-code-flow-authentication.adoc[Authorization code flow] and xref:security-oidc-bearer-token-authentication.adoc[Bearer token] authentication mechanisms.
+
+Well-known OIDC providers such as https://www.keycloak.org/documentation[Keycloak], https://developer.okta.com/[Okta], https://auth0.com/docs/[Auth0] and many xref:security-openid-connect-providers.adoc[social providers] can be configured with ease.
+
+In this tutorial you will learn how https://auth0.com/docs/[Auth0] can be used to secure your Quarkus endpoints.
```

Comment:
Avoid using self-referential statements where possible:
"In this... "

Also, I think line 11 would make a great opening statement, especially on the landing page tile for skimmers:

```suggestion
Learn how to use the Quarkus Security OpenID Connect extension together with https://auth0.com/docs/[Auth0] authorization to secure your API endpoints.
```

Reply:
This line is a more specific statement, we start with saying Quarkus can support all sort of providers and then say that here we'd look at one of them in particular.

But I can try to use your better worded sentence indeed, with a tiny update:

```Learn how to use the Quarkus OpenID Connect extension (`quarkus-oidc`) together with https://auth0.com/docs/[Auth0] OIDC provider to protect your API endpoints.`

Let me try and see if you will like it